### PR TITLE
David/like button

### PR DESCRIPTION
**Pull request by lmy0125**
Worked on the properties and property details page. 
The like button is working with the database right now.
User who like a property is also correctly displayed.

### CODE REVIEW

File: src/components/PropertyCard/PropertyCard.js
Comment (aidandenlinger) [inline src/components/PropertyCard/PropertyCard.js:40]:
```suggestion
					console.error(err);
```

File: src/components/PropertyCard/PropertyCard.js
Comment (aidandenlinger) [inline src/components/PropertyCard/PropertyCard.js:53]:
```suggestion
				.catch((err) => console.error(err));
```

File: src/components/PropertyCard/PropertyCard.js
Comment (aidandenlinger) [inline src/components/PropertyCard/PropertyCard.js:63]: This section took me a little bit to parse.

First, we set the like to the opposite - that makes total sense, the button was just clicked, we want to flip it.

But then we kind of do opposite logic - `if (like)` really means "the button *was* liked, now we unliked it, delete the mapping". When quickly scanning the code, it seems backwards - we liked the property, so delete the mapping? It takes understanding to get that `like` was the PREVIOUS state, not the state right now.
 
Can it instead be something like
`
let like = !like // flip the like
`
and then just use `like`  everywhere? (so `setLike(like)` to the new status and make `if (like)` add the property pref)

File: src/components/PropertyCard/PropertyCard.js
Comment (aidandenlinger) [inline src/components/PropertyCard/PropertyCard.js:82]:
```suggestion
					<FavoriteIcon fontSize="large" sx={ like ? { color: 'red' } : {}} />
```
**THIS IS UNTESTED, PLEASE ACTUALLY TEST THIS BEFORE COMMITTING**

make it clear what the `like` is changing, just the color, instead of repeating ourselves

File: src/components/PropertyCard/PropertyCard.js
Comment (aidandenlinger) [inline src/components/PropertyCard/PropertyCard.js:91]: if we have this, are the three hardcoded avatars below (lines 92-94) needed? if not, can they be deleted?

File: src/pages/property/[uid].js
Comment (aidandenlinger) [inline src/pages/property/[uid].js:84]: same note as earlier, please also update this function. 

if this function is duplicated should it be defined somewhere else and both can import? i don't know where that'd be so i'm okay leaving it but definitely not the greatest practice

File: src/pages/property/[uid].js
Comment (aidandenlinger) [inline src/pages/property/[uid].js:96]: same note as earlier

File: src/pages/property/[uid].js
Comment (aidandenlinger) [inline src/pages/property/[uid].js:127]: not a big fan of writing `fontSize: 40` everywhere, I think the better approach would have been writing your own wrapper class around PrefIcon or something else to have fontSize 40 so we can change all the icons at once. if we decide we want 45 it's gonna take a lot more work than just updating one class. not a big enough deal to block the merge, but if you're copy pasting a ton it's a sign a class can be made

**Review by aidandenlinger — APPROVED** [review]
some nits but overall good. i'll approve because it's overall good but i'd recommend taking a look at the stuff i commented and seeing if there's anything you want to implement, feel free to request another review from me if you tackle them

File: src/components/PropertyCard/PropertyCard.js
Comment (aidandenlinger) [reply src/components/PropertyCard/PropertyCard.js:91]: also we need a task for getting this to actually work fr fr

File: src/components/PropertyCard/PropertyCard.js
Comment (lmy0125) [reply src/components/PropertyCard/PropertyCard.js:82]: This actually works! It is smart code.

File: src/components/PropertyCard/PropertyCard.js
Comment (lmy0125) [reply src/components/PropertyCard/PropertyCard.js:91]: I left them here because the cards won't look so blank without much data. We can remove them anytime.

File: src/pages/property/[uid].js
Comment (lmy0125) [reply src/pages/property/[uid].js:127]: Good observation and suggestion. Fixed.

File: src/components/PropertyCard/PropertyCard.js
Comment (lmy0125) [reply src/components/PropertyCard/PropertyCard.js:63]: Got it. Fixed.

File: src/pages/property/[uid].js
Comment (lmy0125) [reply src/pages/property/[uid].js:84]: In this case I can't find an easy way to arrange for the best practice (more maintainable) since this function requires other three functions and one variable. I will just leave it for now.

File: src/components/PropertyCard/PropertyCard.js
Comment (lmy0125) [reply src/components/PropertyCard/PropertyCard.js:95]: This is actually working, we just don't have enough users and likes to fill up the space right now (if we don't want the placeholders).

**Pull request merged**
